src/Workspaces/Remote/Core/ServiceHubRemoteHostClient.cs

```diff
@@ -98,14 +98,29 @@ public static async Task<RemoteHostClient> CreateAsync(HostWorkspaceServices ser
                     new object?[] { uiCultureLCID, cultureLCID },
                     cancellationToken).ConfigureAwait(false);
 
-                await client.TryInvokeAsync<IRemoteAsynchronousOperationListenerService>(
-                    (service, cancellationToken) => service.EnableAsync(AsynchronousOperationListenerProvider.IsEnabled, listenerProvider.DiagnosticTokensEnabled, cancellationToken),
-                    callbackTarget: null,
-                    cancellationToken).ConfigureAwait(false);
+                if (AsynchronousOperationListenerProvider.IsEnabled && !IsRpsMachine())
+                {
+                    await client.TryInvokeAsync<IRemoteAsynchronousOperationListenerService>(
+                        (service, cancellationToken) => service.EnableAsync(AsynchronousOperationListenerProvider.IsEnabled, listenerProvider.DiagnosticTokensEnabled, cancellationToken),
+                        callbackTarget: null,
+                        cancellationToken).ConfigureAwait(false);
+                }
 
                 client.Started();
                 return client;
             }
+
+            static bool IsRpsMachine()
```

Comment:
Looks like we are going to disable the OOP call and use the in proc client?

Reply:
I wonder do we need to create a bug to track the progress of this.

Reply:
Yes, that is correct.

Reply:
> Looks like we are going to disable the OOP call and use the in proc client?

No, we are disabling the ability of RPS to wait for Roslyn asynchronous operations to complete when those operations run in ServiceHub.